src/booster/mod.rs

```diff
@@ -0,0 +1,54 @@
+use booster::builder::{BoosterBuilder, ParamsNotAdded, TrainDataNotAdded};
+use dataset::DataSet;
+use Error;
+use {InputMatrix, OutputVec};
+
+mod builder;
+mod ffi;
+
+pub struct EvalResult {
+    metric_name: String,
+    score: f64,
+}
+
+pub struct Booster {
+    handle: lightgbm_sys::BoosterHandle,
+    train_data: DataSet,
+    validation_data: Vec<DataSet>,
+}
```

Comment:
Are these meant to be LoadedDatasets? I would assume fit loads the data

What about Boosters that were trained in advance and are loaded from file? What would their train_data and validation_data be? (and does it even make sense to hold onto these potentially huge datasets?)

Reply:
Yes, you're right. This is a refactoring artifact and should just be the Dataset pointers. Now that you say it, it probably doesn't make sense to add them to the booster, if we build them with fit() anyway.